src/transformers/trainer.py

```diff
@@ -404,6 +404,19 @@ def __init__(
         self.args = args
         # Seed must be set before instantiating the model when using model
         enable_full_determinism(self.args.seed) if self.args.full_determinism else set_seed(self.args.seed)
+
+        generator = torch.Generator()
+        if self.args.data_seed is not None:
+            # TODO change min version of accelerate
+            if self.args.accelerator_config.use_seedable_sampler and is_accelerate_available(min_version="0.34.0"):
+                raise RuntimeError(
+                    "data_seed requires accelerate >= 0.34.0, please install the latest version of accelerate"
+                )
+            generator.manual_seed(self.args.data_seed)
+        else:
+            generator.manual_seed(self.args.seed)
+        self.data_sampler_generator = generator
```

Comment:
With the new version what should essentially happen is:

Instead we should modify `create_accelerator_and_postprocess` and modify this chunk of it https://github.com/huggingface/transformers/blob/main/src/transformers/trainer.py#L4860-L4866

With something like:

```python
        if is_accelerate_available("0.28.0"):
            dataloader_config = DataLoaderConfiguration(
                split_batches=accelerator_config.pop("split_batches"),
                dispatch_batches=accelerator_config.pop("dispatch_batches"),
                even_batches=accelerator_config.pop("even_batches"),
                use_seedable_sampler=accelerator_config.pop("use_seedable_sampler"),
            )
            if is_accelerate_available("1.1.0"):
                dataloader_config.data_seed = self.args.data_seed
```
And then in `TrainingArguments.__post_init__` check if a `data_seed` was passed/not the default. If so, do a check for the accelerate version and raise a `NotImplementedError()` telling users to upgrade their accelerate version.